modules/economy.js

```diff
@@ -1,295 +1,255 @@
-                            if (channel) {
-                              channel.send(action.message || 'Nothing', true)
-                              callback(null, true)
-                            } else {
-                              callback(i18n(guild.locale, 'BUYPRODUCT_SENDMESSAGE_ERROR'), false)
-                            }
-                          })
-                        } else {
-                          callback(i18n(guild.locale, 'BUYPRODUCT_SENDMESSAGE_ERROR'), false)
-                        }
-                      }
-                      break
-                    }
-                    case 'giveRole': {
-                      if (Object.prototype.hasOwnProperty.call(action, 'role')) {
-                        guild.roles.fetch(action.role).then(role => {
-                          if (role) {
-                            member.roles.add(role)
-                            callback(i18n(guild.locale, 'BUYPRODUCT_GIVEROLE', { ROLE: role }), true)
-                          } else {
-                            callback(i18n(guild.locale, 'BUYPRODUCT_GIVEROLE_ERROR'), false)
-                          }
-                        })
-                      } else {
-                        callback(i18n(guild.locale, 'BUYPRODUCT_GIVEROLE_ERROR'), false)
-                      }
-                    }
-                  }
+            break
+          }
+          case 'giveRole': {
+            if (Object.prototype.hasOwnProperty.call(action, 'role')) {
+              guild.roles.fetch(action.role).then(role => {
+                if (role) {
+                  member.roles.add(role)
+                  if (callback) callback()
+                } else {
+                  if (callback) callback(Error('ECO_ATI06'))
                 }
-              }
-            })
+              })
+            } else {
+              if (callback) callback(Error('ECO_ATI07'))
+            }
+            break
+          }
+          default: {
+            if (callback) callback()
+            break
           }
-        } else {
-          userInputRequirements.forEach(userInputRequirement => { propertiesString += ` ${userInputRequirement}` })
-          callback(i18n(guild.locale, 'BUYPRODUCT_MISSING_PROPERTY', { PROPERTY: propertiesString }), false)
         }
       }
     }
+  },
+  processMemberInputs: (memberInput, memberInputRequirements, callback) => {
+    const memberInputOrganized = {}
+    const missingInputs = []
+
+    let count = 0
+    memberInputRequirements.forEach(userInputRequirement => {
+      count++
+      memberInputOrganized[userInputRequirement] = 'null'
+      if (memberInputRequirements.length === count) asignValuesToMemberInputOrganized()
+    })
+
+    function asignValuesToMemberInputOrganized () {
+      count = 0
+      memberInput.forEach(input => {
+        count++
+        const getPropertyAndItsValue = input.split(':')
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/alexandrumuntenas/pingu/pullRequest?prid=8657799)
```suggestion
        const getPropertyAndItsValue = input.split(":")
```